enclave_manager/avalon_enclave_manager/worker_kv_delegate.py

```diff
@@ -58,6 +58,17 @@ def cleanup_worker(self, worker_ids=None):
                 result &= self._kv_helper.remove("workers", worker)
         return result
 
+    def cleanup_pool(self, worker_id):
+        """
+        Remove worker pool mapping from datastore. All workers need to
+        register afresh to be a part of the pool.
```

Comment:
`cleanup_pool()` has to be called by both KME and Singleton or only KME? I see `update_worker_map()` called by both Singleton and WPE.

Reply:
In case of singleton, removing and adding would yield the same result.